### PR TITLE
Added some missing offsets for dedicated server.

### PR DESCRIPTION
**Pull request by nukeulater**
Fixed some compiler warnings.
Reorganized some code to follow the new module format (check Camera, UI).
Moved all startup hooks to the appropriate file.
Renamed teamspeak class pointers to be more distinguishable. 
Disabled custom engine initalization until ready.

### CODE REVIEW

**Comment by pnill** [comment]
We didn't want to get into the game engine after the game type had loaded we needed an in between point where if we called the original function it would load everything (similar to on death pre death), honestly getting tired of people changing that hook. The whole deal was supposed to be on map load (which was where it originally was, and what it was originally named)

**Comment by nukeulater** [comment]
I renamed it back (the offsets are the same).

**Comment by iSupersniper** [comment]
Why did you change this?

https://github.com/PermaNulled/cartographer/pull/150/commits/2ee6b41d8e02fd81e4925d7c85a24c3b178c57db

**Comment by nukeulater** [comment]
I moved everything UI related to a folder and created a header with a namespace to centralize UI related stuff.

**Comment by iSupersniper** [comment]
I believe Permanull restructured the code to be on a module base, not a category base.  So instead of having 1 centralized cpp file for all stuff modifying the mainmenu, we have a folder named main menu with each modification it's own module.  Therefore any code that is related to it can be easily found by people who are new to the project and want to help out.  This creates an SDK environment.

**Comment by nukeulater** [comment]
Most of the stuff inside Modules folder is Glitchy's / Yoshi's code, w/ renamed file names and put in folders, categorised by the function they do.
I did the same by creating a folder with all the UI related stuff, based on what element they modify (ex radar, crosshair) and centralize it within the UI header file that has a namespace where we place functions/variables, which is very friendly to work with imo.

**Comment by nukeulater** [comment]
To avoid any further issues/confusion, I sugggest you guys write down some code structure rules to follow.

File: xlive/Globals.cpp [hunk 1]
```diff
@@ -25,9 +25,9 @@ Players* players = new Players();
 bool microphoneEnabled = true;
 std::unordered_map<XUID, BOOL> xuidIsTalkingMap;
 std::unordered_map<XUID, anyID> xuidToTSid;
-IN_ADDR clientMachineAddress = {};
-TSClient* client = NULL;
-TSServer* server = NULL;
+IN_ADDR clientMachineAddress = { };
+TSClient* TS_client = NULL;
```
Comment (shayaantx) [inline xlive/Globals.cpp:29]: why...this naming is completely different with naming of every other variable in this file?

File: xlive/H2MOD.cpp [hunk 2]
```diff
@@ -1912,16 +1881,20 @@ VOID CALLBACK UpdateDiscordStateTimer(HWND hwnd, UINT uMsg, UINT_PTR idEvent, DW
...
-	//HANDLE hThread = CreateThread(NULL, 0, Thread1, NULL, 0, NULL);
-
 	this->Base = (DWORD)game_info.base;
 
 	if (game_info.process_type == H2Types::H2Server)
 	{
 		this->Server = TRUE;
+
+		// Initialize teamspeak server
+		if (TS_server == NULL)
```
Comment (shayaantx) [inline xlive/H2MOD.cpp:1891]: last I checked this happens when we apply hooks for the dedi? has that changed now?

File: xlive/H2MOD/Modules/Camera/FOV/FOV.cpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+void Camera::Tweaks::SetFOV(float field_of_view_degrees) 
+{
+	if (h2mod->Server)
+		return;
+
+	if (field_of_view_degrees > 0.f && field_of_view_degrees <= 110.f)
+	{
+		float current_FOV = *reinterpret_cast<float*>(h2mod->GetBase() + 0x41D984);
+
+		//int res_width = *(int*)(H2BaseAddr + 0xA3DA00); //wip
```
Comment (shayaantx) [inline xlive/H2MOD/Modules/Camera/FOV/FOV.cpp:16]: if this code isn't ready, is this method ready for commit?

if you have stuff not ready, it should be prefixed with a TODO/issue/feature number

**Comment by shayaantx** [comment]
please submit PR's as we discussed, this is too massive of a change

File: xlive/Globals.cpp (hunk 1 above)
Comment (nukeulater) [reply xlive/Globals.cpp:29]: im sure the new naming is better than "server" and "client"

**Pull request closed without merging**